project/cmake/scripts/common/macros.cmake

```diff
@@ -44,7 +44,13 @@ function(core_add_test_library name)
   core_add_library(${name} NO_MAIN_DEPENDS)
   set_target_properties(${name} PROPERTIES EXCLUDE_FROM_ALL 1)
   foreach(src ${SOURCES})
-    set(test_sources ${CMAKE_CURRENT_SOURCE_DIR}/${src} ${test_sources} CACHE STRING "" FORCE)
+    # This will prepend CMAKE_CURRENT_SOURCE_DIR if the path is relative,
+    # otherwise use the absolute path. We can't just prepend
+    # CMAKE_CURRENT_SOURCE_DIR as this source file might be genertaed
+    # in the build directory (in which case it would have been passed
+    # to SOURCES as an absolute path).
```

Comment:
since the context of this comment will be mostly lost after this commit(unless one looks at the history, simply saying:
"resolve absolute path" should be good enough